src/braket/circuits/text_diagram_builders/text_circuit_diagram.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Literal, Union
+
+import braket.circuits.circuit as cir
+from braket.circuits.circuit_diagram import CircuitDiagram
+from braket.circuits.instruction import Instruction
+from braket.circuits.moments import MomentType
+from braket.circuits.result_type import ResultType
+from braket.circuits.text_diagram_builders.text_circuit_diagram_utils import (
+    _add_footers,
+    _categorize_result_types,
+    _compute_moment_global_phase,
+    _group_items,
+    _prepare_qubit_identifier_column,
+    _unite_strings,
+)
+from braket.registers.qubit import Qubit
+from braket.registers.qubit_set import QubitSet
+
+
+class TextCircuitDiagram(CircuitDiagram):
```

Comment:
I know this is just a draft PR but may want to include a docstring for the class later on:
https://amazon-braket-sdk-python--898.org.readthedocs.build/en/898/_apidoc/braket.circuits.text_diagram_builders.text_circuit_diagram.html